depends: Use base system's `sha256sum` utility on FreeBSD

hebasto opened the pull request:
On FreeBSD, the `shasum` utility is provided by the [`perl5`](https://ports.freebsd.org/cgi/ports.cgi?query=%5Eperl5&stype=all&sektion=all) port, which is not part of the base system and must be [installed](https://github.com/hebasto/bitcoin-core-nightly/blob/0e3518579a02e6c79bafc67817136220579a0d5c/.github/workflows/freebsd.yml#L104) separately. Note that this requirement is currently not documented in [`depends/README.md`](https://github.com/bitcoin/bitcoin/blob/master/depends/README.md).

This PR switches to using the [`sha256sum`](https://man.freebsd.org/cgi/man.cgi?query=sha256sum&apropos=0&sektion=0&manpath=FreeBSD+14.2-RELEASE+and+Ports&arch=default&format=html) utility, which is included in the base system.

l0rinc commented on depends/builders/freebsd.mk:
```diff
@@ -1,5 +1,5 @@
 build_freebsd_CXX=clang++
 
-build_freebsd_SHA256SUM = shasum -a 256
+build_freebsd_SHA256SUM = sha256sum
```
Doesn't this apply to [NetBSD](https://github.com/bitcoin/bitcoin/blob/7b06ffce9c50110b475c722918c55a14402346a5/depends/builders/netbsd.mk#L1) as well?

hebasto replied:
No, it doesn't, unfortunately.

NetBSD's [`sha256`](https://man.netbsd.org/sha256.1) is not quite compatible with our depends build syytem.

l0rinc commented:
utACK f0b70fd7a1b1520502f99c516f8e1e3d83368d2c

theuni commented on depends/builders/darwin.mk:
```diff
@@ -6,7 +6,7 @@ build_darwin_STRIP:=$(shell xcrun -f strip)
 build_darwin_NM:=$(shell xcrun -f nm)
 build_darwin_DSYMUTIL:=$(shell xcrun -f dsymutil)
-build_darwin_SHA256SUM=shasum -a 256
+build_darwin_SHA256SUM=sha256sum
```
`sha256sum` has never existed by default on macOS afaik. That's actually why this variable exists in the first place. perl or no, shasum is always there.

It's not present for me on Sonoma.

hebasto replied:
Thanks! This change has been dropped.

l0rinc replied:
hmmm, it was working for me on Sequoia - it likely comes from `coreutils`, right?

theuni replied:
Yes, it's from `coreutils`. But shasum exists out of the box.

hebasto committed:
depends: Use base system's `sha256sum` utility

On FreeBSD, the `shasum` utility is provided by the `perl5` port, which
is not part of the base system and must be installed separately.
Note that this requirement is currently not documented in
`depends/README.md`.

This change switches to using the `sha256sum` utility, which is included
in the base system.

davidgumberg commented:
Tested depends build on a fresh BSD 14.2 install[^1], `sha256sum` is available and `shasum` is not. 

```console
root@freebsd:~ # freebsd-version
14.2-RELEASE
root@freebsd:~ # shasum
-sh: shasum: not found
root@freebsd:~ # sha256sum --version
sha256sum (FreeBSD) 14.2
```

<details> <summary> Full build log</summary>

```console
root@freebsd:~ # fetch https://github.com/bitcoin/bitcoin/archive/refs/heads/master.zip
root@freebsd:~ # sha256sum master.zip
30cadd9f1293505f4e70173d9d1e3f763734773fd7ef943923bcce85e8fc0f7b  master.zip
root@freebsd:~ # tar -xf master.zip
root@freebsd:~ # pkg install -y bash gmake cmake-core curl
root@freebsd:~ # cd bitcoin-master/depends
root@freebsd:~/bitcoin-master/depends # sed -i '' 's/shasum -a 256/sha256sum/g' builders/freebsd.mk
root@freebsd:~/bitcoin-master/depends # gmake -j $(nproc)
# [...]
copying packages: boost libevent bdb sqlite zeromq
to: ~/bitcoin-master/depends/x86_64-unknown-freebsd14.2
To build Bitcoin Core with these packages, pass '--toolchain ~/bitcoin-master/depends/x86_64-unknown-freebsd14.2/toolchain.cmake' to the first CMake invocation.
root@freebsd:~/bitcoin-master/depends # cd ..
root@freebsd:~/bitcoin-master # cmake -B build --toolchain ~/bitcoin-master/depends/x86_64-unknown-freebsd14.2/toolchain.cmake
root@freebsd:~/bitcoin-master # cmake --build build -j $(nproc)
```
</details>

Just to note: `perl5` (and shasum) is installed as a [dependency of git](https://ports.freebsd.org/cgi/ports.cgi?query=^git&stype=all&sektion=all), and I believe `gmake` and `curl` are also required

[^1]:  https://download.freebsd.org/releases/VM-IMAGES/14.2-RELEASE/amd64/Latest/FreeBSD-14.2-RELEASE-amd64.qcow2.xz `75146ff57cb965624630da4e7113db3f54b26807b8e9ab9f5242212abeea0e99  FreeBSD-14.2-RELEASE-amd64.qcow2.xz`

davidgumberg commented on depends/builders/freebsd.mk:
```diff
@@ -1,5 +1,5 @@
 
-build_freebsd_SHA256SUM = shasum -a 256
+build_freebsd_SHA256SUM = sha256sum
 build_freebsd_DOWNLOAD = curl --location --fail --connect-timeout $(DOWNLOAD_CONNECT_TIMEOUT) --retry $(DOWNLOAD_RETRIES) -o
```
```suggestion
build_freebsd_DOWNLOAD = fetch --timeout 180 --retry -o
```

related but probably out-of-scope: freebsd's `fetch` utility could be used to also drop the curl dependency, there is a regression since it is unable to limit number of retries([16-year old patch awaits feedback](https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=131427)), and to set a handshake timeout, but these can be approximated with a single connection timeout
